openstack/dns/v2/associate/requests.go

```diff
@@ -0,0 +1,34 @@
+package associate
+
+import "github.com/chnsz/golangsdk"
+
+type Opts struct {
+	RouterOpts `json:"router" require:"true"`
+}
+
+type RouterOpts struct {
+	RouterID     string `json:"router_id" required:"true"`
+	RouterRegion string `json:"router_region,omitempty"`
+}
+
+func Associate(client *golangsdk.ServiceClient, resolverRuleID string, opts Opts) (r AssociateResult) {
+	b, err := golangsdk.BuildRequestBody(opts, "")
```

Comment:
this will the code clearer

```suggestion
func Associate(client *golangsdk.ServiceClient, resolverRuleID string, opts RouterOpts) (r AssociateResult) {
	b, err := golangsdk.BuildRequestBody(opts, "router")
```